packages/components/src/button/index.native.js

```diff
@@ -80,6 +91,12 @@ export default function Button( props ) {
 		states.push( 'disabled' );
 	}
 
+	const subscriptInactive = getStylesFromColorScheme( styles.subscriptInactive, styles.subscriptInactiveDark );
+
+	const newChildren = Children.map( compact( children ), ( child ) => {
```

Comment:
I'd rather we use a context for this:
```
in Button:
export const ButtonContext = React.createContext( {
  active: false,
  colorScheme: 'light',
} );

export function Button( props ) {

	...

	const buttonContext = {
	  active: props.active,
	  colorScheme: props.colorScheme,
	};

	{ Children.count( children ) &&
		<ButtonContext.Provider value={ buttonContext }>
			{ children }
		</ButtonContext.Provider>
	}

	...

// in IconButton
<ButtonContext.Consumer>
    { ( { colorScheme, active } )  =>
    	<Icon icon={ icon } ariaPressed={ active } colorScheme={ colorScheme } />
</ButtonContext.Consumer>

Reply:
I copied the idea from here: https://github.com/WordPress/gutenberg/blob/82a96dd640354ba5068dee5a10e6195e96ea65bb/packages/components/src/primitives/block-quotation/index.native.js#L15

And the main idea was to clear up `IconButton` from sending the `ariaPressed` or active related states.

I like that we can get enough information to style any child of button, but you are right that this might not be the best option.

I'll give this a try!

Reply:
Thinking about it, I guess using cloneElement should not have any impact on performance, it's just a bit more limited in terms of features. Basically, you'll only be able to access those props from the direct children of the button, whereas a consumer could be added anywhere. If direct children is fine then I guess we can keep it that way.

Reply:
Yes, the main (and only, so far) usage of this is the IconButton where the `Icon` component is a direct child.
So, if you think it's fine to keep this, let's move on 👍 

I restarted a CI job to get the ✅ from travis. 
Thanks for your review!

Reply:
I didn't know about React Context before. Thanks for leaving this comment @Tug !